diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2dc33ef4..cee165481 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,28 +17,17 @@ if (NOT DOLFIN_VERSION_RELEASE)
 endif()
 
 #------------------------------------------------------------------------------
-# Require and use C++11
+# Require and use C++17
 
-# Use C++11
-set(CMAKE_CXX_STANDARD 11)
+# Use C++17
+set(CMAKE_CXX_STANDARD 17)
 
-# Require C++11
+# Require C++17
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # Do not enable compler-specific extensions
 set(CMAKE_CXX_EXTENSIONS OFF)
 
-#------------------------------------------------------------------------------
-# Check compiler version
-
-# Check for GCC version - earlier versions have insuffcient C++11
-# support, or bugs.
-if (CMAKE_COMPILER_IS_GNUCXX)
-  if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8.3)
-    message(FATAL_ERROR "GCC version must be at least 4.8.3 (for sufficient C++11  support and to avoid some compiler bugs). You have version ${CMAKE_CXX_COMPILER_VERSION}")
-  endif()
-endif()
-
 #------------------------------------------------------------------------------
 # Get GIT changeset, if available
 
@@ -322,7 +311,7 @@ if (DOLFIN_ENABLE_MPI AND MPI_C_FOUND)
       URL "http://glaros.dtc.umn.edu/gkhome/metis/parmetis/overview"
       PURPOSE "Enables parallel graph partitioning")
   endif()
-  
+
   if (DOLFIN_ENABLE_SUNDIALS)
     find_package(SUNDIALS 3)
     set_package_properties(SUNDIALS PROPERTIES TYPE OPTIONAL
diff --git a/cmake/templates/UseDOLFIN.cmake.in b/cmake/templates/UseDOLFIN.cmake.in
index bd0d89688..39836179b 100644
--- a/cmake/templates/UseDOLFIN.cmake.in
+++ b/cmake/templates/UseDOLFIN.cmake.in
@@ -12,8 +12,8 @@ if (NOT DOLFIN_USE_FILE_INCLUDED)
     cmake_policy(SET CMP0012 NEW)
   endif()
 
-  # Require and use C++11
-  set(CMAKE_CXX_STANDARD 11)
+  # Require and use C++17
+  set(CMAKE_CXX_STANDARD 17)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
   set(CMAKE_CXX_EXTENSIONS OFF)
 
