diff --git a/dolfin/la/PETScVector.cpp b/dolfin/la/PETScVector.cpp
index 1bfc83f..a785720 100644
--- a/dolfin/la/PETScVector.cpp
+++ b/dolfin/la/PETScVector.cpp
@@ -35,6 +35,8 @@
 #include "PETScVector.h"
 #include "PETScFactory.h"
 
+#include "petscsf.h"
+
 using namespace dolfin;
 
 const std::map<std::string, NormType> PETScVector::norm_types
@@ -750,9 +752,9 @@ void PETScVector::gather(GenericVector& y,
   // Perform scatter
   VecScatter scatter;
 #if PETSC_VERSION_GE(3,11,0)
-  ierr = VecScatterCreateWithData(_x, from, _y.vec(), to, &scatter);
-#else
   ierr = VecScatterCreate(_x, from, _y.vec(), to, &scatter);
+#else
+  ierr = VecScatterCreateWithData(_x, from, _y.vec(), to, &scatter);
 #endif
   CHECK_ERROR("VecScatterCreate");
   ierr = VecScatterBegin(scatter, _x, _y.vec(), INSERT_VALUES,
